### PR TITLE
Feature/lattice 2999 get org studies

### PR DESCRIPTION
**Pull request by solomontang**

### CODE REVIEW

File: src/main/kotlin/com/openlattice/chronicle/services/studies/StudyService.kt [hunk 1]
```diff
@@ -157,6 +157,18 @@ class StudyService(
         private val GET_ORGANIZATION_ID = """
             SELECT ${ORGANIZATION_ID.name} FROM ${ORGANIZATION_STUDIES.name} WHERE ${STUDY_ID.name} = ? LIMIT 1 
         """.trimIndent()
+
+        // How to only get studies that also the current user has READ permissions on?
```
Comment (geekbeast) [inline src/main/kotlin/com/openlattice/chronicle/services/studies/StudyService.kt:161]: You can inner join with permission table ON study_id = acl_key[1]

Comment (geekbeast) [reply src/main/kotlin/com/openlattice/chronicle/services/studies/StudyService.kt:161]: You could also filter by checking authorization on each study but that will be super slow for large number of studies. 

The other way is to invert question and ask which studies that the current user have read access on and the call get studies for those (iirc there's already a getAccessibleObjects(...) that takes securable object type as one of the parameters)

**Review by geekbeast — APPROVED** [review]

File: src/main/kotlin/com/openlattice/chronicle/controllers/StudyController.kt [hunk 2]
```diff
@@ -170,6 +173,45 @@ class StudyController @Inject constructor(
...
+        logger.info("Retrieving studies with organization id $organizationId on behalf of $currentUserId")
+
+        return try {
+            val (flavor, hds) = storageResolver.getDefaultPlatformStorage()
+            ensureVanilla(flavor)
+            val studies = studyService.getOrgStudies(organizationId)
+            AuditedOperationBuilder<Unit>(hds.connection, auditingManager)
+                .audit {
+                    studies.map { study ->
+                        AuditableEvent(
```
Comment (geekbeast) [inline src/main/kotlin/com/openlattice/chronicle/controllers/StudyController.kt:195]: You can now omit a bunch of these variables, if you specify arguments names.

File: src/main/kotlin/com/openlattice/chronicle/services/studies/StudyService.kt (hunk 1 above)
Comment (solomontang) [reply src/main/kotlin/com/openlattice/chronicle/services/studies/StudyService.kt]: updated sql to join with permissions table

File: src/main/kotlin/com/openlattice/chronicle/services/studies/StudyService.kt [hunk 3]
```diff
@@ -184,6 +185,35 @@ class StudyService(
...
+         * include list of all organizationIds each study is a part of.
+         * sort by most recently created
+         * 1. current user principal id
+         * 2. organization id
+        */
+        private val GET_ORG_STUDIES_SQL = """
+            SELECT ${STUDIES.name}.*, org_studies.${ORGANIZATION_IDS.name}
+            FROM ${STUDIES.name}
+                INNER JOIN ${PERMISSIONS.name}
+                ON ${STUDIES.name}.${STUDY_ID.name} = ANY(${PERMISSIONS.name}.${ACL_KEY.name})
```
Comment (geekbeast) [inline src/main/kotlin/com/openlattice/chronicle/services/studies/StudyService.kt:201]: This won't matter for this purpose but noting it should be ```ACL_KEY.name[1]``` as the reason that it is an array is that so there is somewhat of hierarchy (only used by roles atm).

**Comment by geekbeast** [comment]
Looks good can we get a test for this? You can use clientUser1 and clientUser2 to create two orgs and two studies and make sure that things are working as expected.

**Review by geekbeast — APPROVED** [review]

**Pull request merged**